django/db/models/sql/query.py

```diff
@@ -1613,18 +1613,34 @@ def trim_joins(self, targets, joins, path):
             self.unref_alias(joins.pop())
         return targets, joins[-1], joins
 
+    @classmethod
+    def _gen_col_aliases(cls, exprs):
+        for expr in exprs:
+            if isinstance(expr, Col):
+                yield expr.alias
+            else:
+                yield from cls._gen_col_aliases(expr.get_source_expressions())
+
     def resolve_ref(self, name, allow_joins=True, reuse=None, summarize=False, simple_col=False):
         if not allow_joins and LOOKUP_SEP in name:
             raise FieldError("Joined field references are not permitted in this query")
-        if name in self.annotations:
+        annotation = self.annotations.get(name)
+        if annotation is not None:
+            if not allow_joins:
```

Comment:
Remove a level of indentation by flattening this if statement into the previous.

Reply:
:thinking: As far as I'm concerned I will not change a level of indentation:
```python
if name in self.annotations:
    if not allow_joins:
        annotation = self.annotations[name]
```

Reply:
Maybe you missed `if summarize` branch (below).

Reply:
Ah, sneaky 🤗 Too early in the morning!